ee/localserver/krypto-ec-middleware.go

```diff
@@ -71,16 +73,23 @@ func (cmdReq v2CmdRequestType) CallbackReq() (*http.Request, error) {
 }
 
 type kryptoEcMiddleware struct {
-	localDbSigner crypto.Signer
-	counterParty  ecdsa.PublicKey
-	slogger       *slog.Logger
+	localDbSigner                     crypto.Signer
+	counterParty                      ecdsa.PublicKey
+	slogger                           *slog.Logger
+	presenceDetector                  presenceDetector
+	presenceDetectionLock             sync.Mutex
+	presenceDetectionCallbackInterval time.Duration
+	timestampValidityRange            int64
 }
 
-func newKryptoEcMiddleware(slogger *slog.Logger, localDbSigner crypto.Signer, counterParty ecdsa.PublicKey) *kryptoEcMiddleware {
+func newKryptoEcMiddleware(slogger *slog.Logger, localDbSigner crypto.Signer, counterParty ecdsa.PublicKey, presenceDetector presenceDetector) *kryptoEcMiddleware {
 	return &kryptoEcMiddleware{
-		localDbSigner: localDbSigner,
-		counterParty:  counterParty,
-		slogger:       slogger.With("keytype", "ec"),
+		localDbSigner:                     localDbSigner,
+		counterParty:                      counterParty,
+		slogger:                           slogger.With("keytype", "ec"),
+		presenceDetector:                  presenceDetector,
+		timestampValidityRange:            timestampValidityRange,
+		presenceDetectionCallbackInterval: 30 * time.Second,
```

Comment:
Not for this PR -- but maybe for the future we'd want this set in the knapsack via the `agent_flags` subsystem from the control server, so K2 could reduce the callback interval if needed?

Reply:
What's it used for?

Reply:
`on an interval (30s currently) send a callback letting server side know we're still waiting on user`

Reply:
updated name based on Becca recommendation add comments